x-pack/plugins/cases/server/services/attachments/operations/get.ts

```diff
@@ -247,4 +250,88 @@ export class AttachmentGetter {
       },
     };
   }
+
+  public async getFileAttachments({
+    caseId,
+    fileIds,
+  }: {
+    caseId: string;
+    fileIds: string[];
+  }): Promise<Array<SavedObject<CommentAttributes>>> {
+    try {
+      this.context.log.debug('Attempting to find file attachments');
+
+      /**
+       * This is making a big assumption that a single file service saved object can only be associated within a single
+       * case. If a single file can be attached to multiple cases it will complicate deleting a file.
+       *
+       * The file's metadata would have to contain all case ids and deleting a file would need to removing a case id from
+       * array instead of deleting the entire saved object in the situation where the file is attached to multiple cases.
+       */
+      const references = fileIds.map((id) => ({ id, type: FILE_SO_TYPE }));
+
+      /**
+       * In the event that we add the ability to attach a file to a case that has already been uploaded we'll run into a
+       * scenario where a single file id could be associated with multiple case attachments. So we need
+       * to retrieve them all.
+       */
+      const finder =
+        this.context.unsecuredSavedObjectsClient.createPointInTimeFinder<AttachmentAttributesWithoutRefs>(
```

Comment:
I didn't know this `createPointInTimeFinder` API so I was reading the `saved_objects_client` docs.

Considering we have the file ids `references` already are we really anticipating:

> page through large sets of saved objects. We strongly recommend using this API for
> any `find` queries that might return more than 1000 saved objects, [...]

?

Reply:
Yeah this is more of a safety measure. It's probably unlikely that we'd get more than 100 (or 50 if we lower the max number of files on a case). We could use the find api to do the same thing but it's limited to 10k results where as this will get all of them.